tasks/eva_3371/src/main/groovy/eva3371/eva3371_detect_unnormalized_indels.groovy

```diff
@@ -0,0 +1,211 @@
+    def impactedVariantTypes = [VariantType.INS.toString(), VariantType.DEL.toString(),
+                                VariantType.INDEL.toString()]
+    def allSveCursors = [sveClass, dbsnpSveClass].collect {
+        new RetryableBatchingCursor(where("seq").is(assembly), env.mongoTemplate, it)}
+    // We are going to mimic EVA extractor logic (just the processor and the writer parts) here: https://github.com/EBIvariation/eva-accession/blob/33af9bab89219c5071ea77c3c8cd3a1c031a78f3/eva-remapping-get-source/src/main/java/uk/ac/ebi/eva/remapping/source/configuration/batch/steps/ExportSubmittedVariantsStepConfiguration.java#L79-L79
+    // to minimize the pain of writing the SVEs above to VCF
+    def sveToVariantContextProcessor =
+            env.springApplicationContext.getBean(RemappingExtractBeanNames.SUBMITTED_VARIANT_PROCESSOR,
+                    ItemProcessor<SubmittedVariantEntity, VariantContext>.class)
+    def outputVCFFileName = Paths.get(outputDir + "/" + "${assembly}_before_norm.vcf").toString()
+    // See here: https://github.com/EBIvariation/eva-accession/blob/33af9bab89219c5071ea77c3c8cd3a1c031a78f3/eva-remapping-get-source/src/main/java/uk/ac/ebi/eva/remapping/source/configuration/batch/io/VariantContextWriterConfiguration.java#L34-L34
+    def variantContextWriter = new VariantContextWriter(Paths.get(outputVCFFileName), assembly)
+    variantContextWriter.open(new ExecutionContext())
+
+    allSveCursors.each{it.each {allSves ->
+        def impactedSves = allSves.findAll {sve ->
+            def type = VariantType.SNV.toString()
+            try {
+                type = EVAObjectModelUtils.toClusteredVariant(sve).getType().toString()
+            }
+            catch (IllegalArgumentException exception) {
+            }
+            return impactedVariantTypes.contains(type)
+        }
+        variantContextWriter.write(impactedSves.collect{sveToVariantContextProcessor.process(it)}
+                .findAll{Objects.nonNull(it)})
+    }}
+    variantContextWriter.close()
+    outputVCFFileName = sortAndIndexVcf(outputVCFFileName, config)
+    return Paths.get(outputVCFFileName).toAbsolutePath().toString()
+}
+
+String getNormalizedVcf (String preNormalizedVCF, String customFastaFilePath, config) {
+    def postNormalizedVCF = preNormalizedVCF.replace("before_norm", "after_norm")
+    runProcess("${config.executable.bcftools} norm -c w -f ${customFastaFilePath} ${preNormalizedVCF} " +
+            "-o ${postNormalizedVCF} -Oz")
+    runProcess("${config.executable.tabix} -f ${postNormalizedVCF}")
+    return postNormalizedVCF
+}
+
+String getVcfWithChangedVariants(String preNormalizedVCF, String postNormalizedVCF, config) {
+    def changedVCF = postNormalizedVCF.replace("after_norm", "changed_after_norm")
+    runProcess("${config.executable.bcftools} isec -w1 -C ${postNormalizedVCF} ${preNormalizedVCF} " +
+            "-o ${changedVCF} -Oz")
```

Comment:
Maybe using `--old-rec-tag` which record the variant that have been normalised would allow you to detect them more easily.
https://samtools.github.io/bcftools/bcftools.html#norm